### PR TITLE
Reviewer MIRW: Mangelwurzel leaves the Req URI alone on in dialog requests

### PR DESCRIPTION
**Pull request by graemerobertson**
Hi Matt,

Please can you review my change to mangelwurzel so that it doesn't mangle the Request URI on in-dialog requests? As discussed, mangling the Request URI on in-dialog requests because we've mangled it on the initial request doesn't make sense. We may wish to mangle it for other reasons on in-dialog requests, but right now we don't have a well defined use case for this.

I've tested by running the UTs and deploying on a system.

I've updated the design.

Thanks,
Graeme

### CODE REVIEW

File: sprout/mangelwurzel/mangelwurzel.cpp [hunk 1]
```diff
@@ -264,11 +263,6 @@ void MangelwurzelTsx::on_rx_in_dialog_request(pjsip_msg* req)
     mangle_dialog_identifiers(req, pool);
   }
 
-  if (_config.req_uri)
-  {
-    mangle_req_uri(req, pool);
-  }
-
   if (_config.contact)
```
Comment (mirw) [inline sprout/mangelwurzel/mangelwurzel.cpp]: Why do we mangle the Contact URI if we're not mangling the request URI?  Is the key point here that we should be enabling Contact URI mangling whenever we're mangling the request URI?

**Comment by mirw** [comment]
One comment about whether this is necessary.  Apart from that, it's fine.

**Comment by graemerobertson** [comment]
Nice spot. I've completely changed this now, as discussed. Please can you re-review? If you're happy with this, I'll update the docs to reflect the removal of the contact option.

I've tested live and run the UTs.

**Comment by mirw** [comment]
As discussed, you should remove the `contact` field from the `Config` class... and please check for other instances.

Other than that, looks fine.

**Pull request merged**